iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java

```diff
@@ -103,26 +105,125 @@ public static double readDoubleAttributeFromMinimumVersion(String rootElementNam
+            try {
+                context.getWriter().writeAttribute(attributeName, Boolean.toString(value));
+            } catch (XMLStreamException e) {
+                throw new UncheckedXmlStreamException(e);
+            }
+        });
+    }
+
+    /**
+     * Write a double attribute from a given minimum IIDM-XML version if its value is defined.<br>
+     * If the context's IIDM-XML version is strictly older than the given minimum IIDM-XML version, the attribute's value <b>should be undefined i.e. NaN</b>
+     * (else an exception is thrown).
+     */
+    public static void writeDoubleAttributeFromMinimumVersion(String rootElementName, String attributeName, double value,
+                                                              ErrorMessage type, IidmXmlVersion minVersion, NetworkXmlWriterContext context) {
+        writeDoubleAttributeFromMinimumVersion(rootElementName, attributeName, value, Double.NaN, type, minVersion, context);
+    }
+
+    /**
+     * Write a double attribute from a given minimum IIDM-XML version if its value is defined.<br>
      * If the context's IIDM-XML version is strictly older than the given minimum IIDM-XML version, the attribute's value <b>should be default</b>
      * (else an exception is thrown).
      */
     public static void writeDoubleAttributeFromMinimumVersion(String rootElementName, String attributeName, double value, double defaultValue,
-                                                              ErrorMessage type, IidmXmlVersion minVersion, NetworkXmlWriterContext context) throws XMLStreamException {
+                                                              ErrorMessage type, IidmXmlVersion minVersion, NetworkXmlWriterContext context) {
+        writeAttributeFromMinimumVersion(rootElementName, attributeName, !Objects.equals(value, defaultValue), type, minVersion, context, () -> {
+            try {
+                XmlUtil.writeDouble(attributeName, value, context.getWriter());
+            } catch (XMLStreamException e) {
+                throw new UncheckedXmlStreamException(e);
+            }
+        });
+    }
+
+    private static void writeAttributeFromMinimumVersion(String rootElementName, String attributeName, boolean isNotDefaultValue,
+                                                         ErrorMessage type, IidmXmlVersion minVersion, NetworkXmlWriterContext context,
+                                                         Runnable write) {
         if (context.getVersion().compareTo(minVersion) >= 0) {
-            XmlUtil.writeDouble(attributeName, value, context.getWriter());
+            write.run();
         } else {
-            assertMinimumVersionIfNotDefault(!Objects.equals(value, defaultValue), rootElementName, attributeName, type, minVersion, context);
+            assertMinimumVersionIfNotDefault(isNotDefaultValue, rootElementName, attributeName, type, minVersion, context);
         }
     }
 
```

Comment:
It seems to me that this piece of development is necessary to the addition of regulations on shunts, because it's the first case like this maybe, but it's actually a modification that could impact other equipments... Maybe it should have been pushed separately? Anyway now that it's here, it should probably be mentioned in the PR's main text just to keep track of it, what do you think?

Reply:
Git is really weird in its displaying of differences, the differences here do not apply to a pre-existing method, it is a new one. It should not result in any behavior change (unless I made a mistake) which is why I did not mention it. Basically, I refactored `writeDoubleAttributeFromMinimumVersion` by implementing a private intermediary method that takes a boolean which will be the result of `!Objects.equals(value, defaultValue)` for double values.